app/uk.gov.hmrc.helptosaveproxy/connectors/NSIConnector.scala

```diff
@@ -216,8 +218,11 @@ class NSIConnectorImpl @Inject()(
         .map[Either[String, HttpResponse]] { response ⇒
           val time = timeContext.stop()
           response.status match {
-            case Status.OK | Status.BAD_REQUEST ⇒
-              logger.debug(s"queryAccount resource: $resource, response: ${response.body}")
+            case Status.OK ⇒
+              logger.debug(s"queryAccount resource: $resource, response: ${response.body} and returned status: 200 OK.")
+              Right(response)
+            case Status.BAD_REQUEST ⇒
+              logger.debug(s"queryAccount resource: $resource, response: ${response.body} and returned status: 400 bad request.")
```

Comment:
The log for the bad request case needs to be at INFO or better at WARN, Production is never generally configured to log at DEBUG.

It would seem prudent to use maskNino.

Do you know if the body gets logged by help-to-save-proxy if the other case returns a Left?